src/ParserRuleContext.ts

```diff
@@ -46,7 +46,26 @@ export class ParserRuleContext extends RuleContext {
 	 *  operation because we don't the need to track the details about
 	 *  how we parse this rule.
 	 */
-	children?: ParseTree[];
+
+
+	_children?: Array<ParseTree>;
+	static _emptyChildren = [];
+
+	/** Access children for read
+	 * Avoids creation of extra empty arrays by using a shared _emptyChildren array
+	 */
+	get children() : ReadonlyArray<ParseTree> {
+		let result = this._children;
+		return result ? result : ParserRuleContext._emptyChildren;
```

Comment:
You could also make _children not allowing undefined, which would avoid the need of _emptyChildren. There is no need to allow _children to be undefined. Being empty serves the same purpose.